src/components/Input.js

```diff
@@ -49,11 +68,18 @@ class Input extends React.PureComponent {
             onFocus={onFocus}
             onKeyPress={onKeyPress}
             onChange={onChange}
+            ref={(input) => { this.input = input; }}
           />
-          {error && <ErrorIcon className="drui-inputError__errorIcon" />}
+          {initialType === 'password' && !error && (
+              type === 'text'
+                ? <PublicIcon className="drui-input__icon drui-input__passwordVisibilityIcon" onClick={this.togglePasswordVisibility} />
```

Comment:
Please, move props declarations to new lines, so we won't have such a long single lines od code